dandiapi/analytics/tasks/__init__.py

```diff
@@ -80,10 +81,19 @@ def process_s3_log_file_task(s3_log_key: str) -> None:
                 logger.info('Already processed log file %s', s3_log_key)
             return
 
+        # we need to store all of the fully hydrated blob objects in memory in order to use
+        # bulk_update, but this turns out to not be very costly. 1,000 blobs use about ~8kb
+        # of memory.
+        asset_blobs = []
+
+        # batch the blob queries to avoid a large WHERE IN clause
+        for batch in batched(download_counts, 1_000):
+            asset_blobs += AssetBlob.objects.filter(blob__in=batch)
+
+        for asset_blob in asset_blobs:
+            asset_blob.download_count = F('download_count') + download_counts[asset_blob.blob]
```

Comment:
What does using `F('download_count')` do here? As opposed to just writing:

```python
asset_blob.download_count += download_counts[asset_blob.blob]
```

Reply:
```sql
-- assuming download_count is already 20 and we're adding 10

-- using F('download_count')
UPDATE asset_blob SET download_count = download_count + 10;

-- without
UPDATE asset_blob SET download_count = 30;
```

The latter is susceptible to the lost update problem.